core/src/main/java/org/apache/calcite/sql/fun/SqlCastFunction.java

```diff
@@ -168,7 +182,7 @@ public SqlCastFunction() {
   }
 
   @Override public SqlSyntax getSyntax() {
-    return SqlSyntax.SPECIAL;
+    return this.kind == SqlKind.CAST ? SqlSyntax.SPECIAL : SqlSyntax.FUNCTION;
```

Comment:
if SAFE_CAST requires AS, it's not function syntax

Reply:
+1, SAFE_CAST is a special SqlSyntax